client/landing/gutenboarding/onboarding-block/style-preview/preview.tsx

```diff
@@ -16,14 +16,49 @@ import { isEnabled } from 'config';
 type Design = import('../../stores/onboard/types').Design;
 type Font = import('../../constants').Font;
 type SiteVertical = import('../../stores/onboard/types').SiteVertical;
+import { fontPairings } from '../../constants';
+
+function getFontsLoadingHTML() {
+	const baseURL = 'https://fonts.googleapis.com/css2';
+
+	// matrix: regular,bold * regular,italic
+	const axis = 'ital,wght@0,400;0,700;1,400;1,700';
+
+	const head = fontPairings.reduce( ( acc, pairing ) => {
+		const query = [
+			`family=${ encodeURI( pairing.headings ) }:${ axis }`,
+			`family=${ encodeURI( pairing.base ) }:${ axis }`,
+		];
+		const l = document.createElement( 'link' );
+		l.rel = 'stylesheet';
+		l.type = 'text/css';
+		l.href = `${ baseURL }?${ query.join( '&' ) }&display=swap`;
+		acc += l.outerHTML;
+		return acc;
+	}, '' );
+
+	// Chrome doesn't load the fonts in memory until they're actually used,
+	// this keeps the fonts used and ready in memory.
+	const fontsHolders = fontPairings.reduce( ( acc, pairing ) => {
+		Object.values( pairing ).forEach( ( font ) => {
+			const fontHolder = document.createElement( 'div' );
+			fontHolder.style.fontFamily = font;
+			fontHolder.innerHTML = 'Placeholder';
```

Comment:
Is it necessary for the elements to contain text? Just curious.

Reply:
Yes! I was amazed too!